shared/js/contacts/import/utilities/overlay.js

```diff
@@ -1,40 +1,151 @@
-'use strict';
-
-var utils = window.utils || {};
-(function() {
-  utils.overlay = {};
-
-  var overlay = document.querySelector('#loading-overlay'),
-      statusContainer = overlay.querySelector('p[role="status"]'),
-      progressActivity = document.querySelector('#progress-activity'),
-      progressTitle = document.querySelector('#progress-title'),
-      progressElement = document.querySelector('#progress-element'),
-      progressMsg = document.querySelector('#progress-msg'),
-      menu = document.querySelector('#loading-overlay menu'),
-      cancelButton = document.querySelector('#cancel-overlay');
-
-  // Constructor for the progress element
-  function ProgressBar(pMsgId, pClass) {
-    var counter = 0;
-    var total = 0;
-    var progressTextId = pMsgId || 'genericProgress';
-    var clazz = pClass;
+/* global LazyLoader, HtmlImports */
 
+(function(exports) {
+  'use strict';
+
+  var counter = 0;
+  var total = 0;
+  var progressTextId;
+  var progressType;
+
+  var overlay,
+      statusContainer,
+      progressActivity,
+      progressTitle,
+      progressElement,
+      progressMsg,
+      menu,
+      cancelButton,
+      _cancelCB,
+      isAnimationPlaying = false,
+      isShown = false;
+
+  var overlayReady,
+      _reject,
+      isLoading = false;
+
+  var link = document.createElement('link');
+  link.setAttribute('rel', 'import');
+  link.setAttribute('href', '/shared/elements/contacts/overlay.html');
+
+  function cacheElements() {
+    statusContainer = overlay.querySelector('p[role="status"]'),
+    progressActivity = document.querySelector('#progress-activity'),
+    progressTitle = document.querySelector('#progress-title'),
+    progressElement = document.querySelector('#progress-element'),
+    progressMsg = document.querySelector('#progress-msg'),
+    menu = document.querySelector('#loading-overlay menu'),
+    cancelButton = document.querySelector('#cancel-overlay');
     progressElement.setAttribute('value', 0);
```

Comment:
I would move this out of the cacheElements, as it's an action on the cached elements (not caching anything itself)